src/features/leda-nft/store/leda-nft.slice.ts

```diff
@@ -40,8 +40,63 @@ const ledaNftSlice = createSlice({
   },
 });
 
+// TODO: Change this name to selectNftState
 export const selectState = (state: RootState) => state.ledaNft;
 
+export const selectAllItems = (state: RootState) => state.ledaNft.items;
+
+export const selectFilteredItems = createSelector(
+  selectAllItems,
+  (
+    _: unknown,
+    author: string,
+    title: string,
+    description: string,
+    priceFrom: number,
+    priceTo: number,
+    likesDirection: string
+  ) => ({
+    author,
+    title,
+    description,
+    priceFrom,
+    priceTo,
+    likesDirection,
+  }),
+  (items: Item[], { author, title, description, priceFrom, priceTo, likesDirection }) => {
+    let filteredItems = [...items];
+    if (author && author !== 'all') {
+      // TODO: The logic is working fine, but we should change the data from the user and add a username
+      filteredItems = filteredItems.filter((item) => item.owner.address === author);
+    }
+    if (title && title !== 'all') {
+      filteredItems = filteredItems.filter((item) => item.name.includes(title));
+    }
+    if (description && description !== 'all') {
+      filteredItems = filteredItems.filter((item) => item.description.includes(description));
+    }
+    if (priceFrom > 0 && priceTo > 0) {
+      // TODO: Get the most expensive item from the store and set it as priceRange.to
+      filteredItems = filteredItems.filter(
+        (item) => item.price > priceFrom && item.price < priceTo
+      );
+    }
+
+    if (likesDirection && likesDirection !== '') {
+      if (likesDirection === 'asc') {
+        // from least liked to most
```

Comment:
I feel like the code is self descriptive. You typically add comments when the code you are writing is difficult to understand. If your code is difficult to understand you are in trouble young man :D. So, no comments needed ;)